geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/commands/ClearCommand.java

```diff
@@ -0,0 +1,80 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.cli.commands;
+
+import static org.apache.geode.management.internal.cli.commands.DataCommandsUtils.callFunctionForRegion;
+
+import java.util.Set;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.springframework.shell.core.annotation.CliCommand;
+import org.springframework.shell.core.annotation.CliOption;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Region;
+import org.apache.geode.distributed.DistributedMember;
+import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.management.cli.CliMetaData;
+import org.apache.geode.management.cli.ConverterHint;
+import org.apache.geode.management.cli.GfshCommand;
+import org.apache.geode.management.internal.cli.domain.DataCommandRequest;
+import org.apache.geode.management.internal.cli.domain.DataCommandResult;
+import org.apache.geode.management.internal.cli.functions.DataCommandFunction;
+import org.apache.geode.management.internal.cli.result.model.ResultModel;
+import org.apache.geode.management.internal.i18n.CliStrings;
+import org.apache.geode.security.ResourcePermission.Operation;
+import org.apache.geode.security.ResourcePermission.Resource;
+
+public class ClearCommand extends GfshCommand {
+  public static final String REGION_NOT_FOUND = "Region <%s> not found in any of the members";
+
+  @CliMetaData(relatedTopic = {CliStrings.TOPIC_GEODE_DATA, CliStrings.TOPIC_GEODE_REGION})
+  @CliCommand(value = {CliStrings.CLEAR}, help = CliStrings.CLEAR_HELP)
+  public ResultModel clear(
+      @CliOption(key = {CliStrings.CLEAR_REGION_NAME}, mandatory = true,
+          help = CliStrings.CLEAR_REGION_NAME_HELP,
+          optionContext = ConverterHint.REGION_PATH) String regionPath) {
+
+    Cache cache = getCache();
+
+    authorize(Resource.DATA, Operation.WRITE, regionPath);
+
+
+    Region region = cache.getRegion(regionPath);
+    DataCommandFunction clearfn = new DataCommandFunction();
+    DataCommandResult dataResult;
+    if (region == null) {
+      Set<DistributedMember> memberList = findAnyMembersForRegion(regionPath);
+
+      if (CollectionUtils.isEmpty(memberList)) {
+        return new ResultModel().createError(String.format(REGION_NOT_FOUND, regionPath));
+      }
+
+      DataCommandRequest request = new DataCommandRequest();
+      request.setCommand(CliStrings.REMOVE);
+      request.setRemoveAllKeys("ALL");
+      request.setRegionName(regionPath);
+      dataResult = callFunctionForRegion(request, clearfn, memberList);
+    } else {
+      dataResult = clearfn.remove(null, null, regionPath, "ALL",
```

Comment:
The remove function already has logic for 'clearing' a region (since you can remove --all) so a clear is just a remove with the all flag set. Anil and I discussed this as well and I refactored the remove all code into a 'clear()' method on the function class so it was more obvious where the code for clear was being performed but there were challenges in implementing a new clear function and it would also have involved duplicate code.